Client/Frontend/Browser/TabManager.swift

```diff
@@ -642,6 +656,10 @@ extension TabManager : WKNavigationDelegate {
         UIApplication.sharedApplication().networkActivityIndicatorVisible = true
     }
 
+    func webView(webView: WKWebView, didCommitNavigation navigation: WKNavigation!) {
+        noImageModePrefDidChange()
```

Comment:
From what I can tell, this toggles the `noImageMode` for every tab, every time the front `webView` starts receiving content.

I suggest you move the code to talk to the WebView  into the tab or the `tabHelper`, the use `tabManager.tab[webView]` to find the tab.